generic3g/vertical/ModelVerticalGrid.F90

```diff
@@ -127,9 +128,9 @@ subroutine get_coordinate_field(this, field, coupler, standard_name, geom, typek
        integer :: i
 
        v_pt = VirtualConnectionPt(state_intent='export', short_name=this%variants%of(1))
-       goal_spec = FieldSpec(geom=geom, vertical_grid=this, vertical_dim_spec=VERTICAL_DIM_EDGE, &
-            typekind=typekind, standard_name=standard_name, units=units, &
-            ungridded_dims=UngriddedDims())
+       goal_spec = FieldSpec( &
+            geom=geom, vertical_grid=this, vertical_dim_spec=vertical_dim_spec, &
+            typekind=typekind, standard_name=standard_name, units=units, ungridded_dims=UngriddedDims())
```

Comment:
What I think is wanted here is that you select a different `short_name` depending on `vertical_dim_spec`.    Then it probably (hopefully) is not needed in `goal_spec`.